### ISSUE TITLE
roslint doesn't seem to check function name style

### ISSUE THREAD

**Issue by spmaniato** [ros/roslint#55]
This is a follow-up from http://answers.ros.org/question/246677

According to the [ROS C++ style guide](http://wiki.ros.org/CppStyleGuide#Function_.2BAC8_Methods), function names should be `camelCased` (lower-case first letter), not `CamelCased` (capital-case first letter). However, roslint doesn't seem to complain about `CamelCased` function names. Actually, it doesn't complain even if a function name is `under_scored` 🤔 

Here's a proof of concept:

```cpp
void my_silly_test() {
    return;
}
```

And here's the output of `catkin build --no-deps <package> --make-args roslint_<package>`:

```
when starting a new scope, { should be on a line by itself  [whitespace/braces]
```

It correctly complains about the scope delimiter, but not the function name.

**Comment by mikepurvis** [comment]
I'd be delighted for a PR to patch in roslint fails for this. :)
